src/main/java/org/opensrp/web/utils/InitialFormConfigUploadUtil.java

```diff
@@ -0,0 +1,392 @@
+				.type(MultipartBuilder.FORM)
+				.addFormDataPart("form_version", formVersion)
+				.addFormDataPart("form_identifier", fileName)
+				.addFormDataPart("form_name", title)
+				.addFormDataPart("form", fileName, RequestBody
+						.create(MediaType.parse(fileContentType), Files.readAllBytes(ymlFile.toPath())));
+
+		//.addPart(RequestBody.create(MediaType.parse("application/x-yaml"), Files.readAllBytes(ymlFile.toPath())));
+
+		if (formRelation != null) {
+			multipartBuilder.addFormDataPart("form_relation", formRelation);
+		}
+
+		return multipartBuilder.build();
+	}
+
+
+	public void performUpload() throws IOException {
+		/*
+		********************************
+		 THESE VALUES SHOULD BE CHANGED
+		********************************
+		*/
+
+		// Path to client project assets folder with json.form
+		String assetsFilePath = "/home/ona-kigamba/Documents/Projects/OpenSRP/opensrp-client-reveal/opensrp-reveal/src/main/assets/";
+		// Initial form version
+		String formVersion = "0.0.1";
+
+		// Base URL for the server to which the forms will be uploaded
+		String baseUrl = "http://192.168.56.103:8082";
+
+		// Authentication details for the server
+		String username = "username";
+		String password = "password";
+		String encoding = new String(java.util.Base64.getEncoder().encode((username + ":" + password).getBytes()));
+
+		String clientFormUrl = baseUrl + "/opensrp/rest/clientForm";
+		String manifestUrl = baseUrl + "/opensrp/rest/manifest";
+		String basicAuthValue = "Basic " + encoding;
+		String appId = "org.smartregister.random";
+		String appVersion = "0.0.1";
+
+		/*
+		********************************
+			END OF CHANGEABLE VALUES
+		********************************
+		*/
+
+		File file = new File(assetsFilePath);
+		HashMap<String, String> ruleFileRelations = new HashMap<>();
+		HashMap<String, String> subFormFileRelations = new HashMap<>();
+		HashSet<String> jsonFormFiles = new HashSet<>();
+		HashMap<String, String> formLabels = new HashMap<>();
+
+		// JSONPath Configuration
+		Configuration conf = Configuration.defaultConfiguration()
+				.addOptions(Option.DEFAULT_PATH_LEAF_TO_NULL);
+
+		if (file.exists() && file.isDirectory()) {
+			File[] files = file.listFiles();
+
+			for (File fileD: files) {
+				String fileName = fileD.getName();
+				if (fileD.isDirectory()) {
+					System.out.println(fileD.getName() + " @ " + fileD.getAbsolutePath());
+					if (fileName.equals("rule")) {
+						File[] ruleFiles = fileD.listFiles();
+						if (ruleFiles != null) {
+							for (File ruleFile : ruleFiles) {
+								String ruleFileName = ruleFile.getName();
+								if (ruleFileName.endsWith(".yml")) {
+									System.out.println(
+											String.format("RULE FILE %s @ %s", ruleFileName, ruleFile.getAbsolutePath()));
+									if (!ruleFileRelations.containsKey(ruleFileName)) {
+										ruleFileRelations.put(fileName, null);
+									}
+								}
+							}
+						}
+					} else if (fileName.equals("json.form")) {
```

Comment:
Can we make this configurable in case an implementation has used different form locations? It might be very rare though